Labyrinth/src/Physics/Board.java

```diff
@@ -143,6 +147,21 @@ public Tile insert(Tile t, Point location, int direction){
 		return result;
 	}
 	
+	/**
+	 * Finds the furthest tile from the cursorLocation in the direction insertDirection.
+	 * @param insertDirection
+	 * @param cursorLocation
+	 * @return
+	 */
+	public Point getFurthestTile(int insertDirection, Point cursorLocation) {
+		Point newCoords = Directions.move(cursorLocation,insertDirection);
+		Tile nextTile = this.tileAt(newCoords);
+		if(nextTile != null) {
+			return this.getFurthestTile(insertDirection, newCoords);
```

Comment:
: ) 

Recursion mastery! 

(No one at work uses recursion, it's nice to see here : )